service/frontend/api/handler.go

```diff
@@ -495,11 +455,6 @@ func (wh *WorkflowHandler) PollForActivityTask(
 		return nil, validate.ErrIdentityTooLong
 	}
 
-	if ok := wh.allow(ratelimitTypeWorker, pollRequest); !ok {
-		// pollers exponentially back off up to 10s
-		return nil, createServiceBusyError()
-	}
```

Comment:
so far I think the only concerning point is that this check is done earlier:
```go
	domainName := pollRequest.GetDomain()
	if domainName == "" {
		return nil, validate.ErrDomainNotSet
	}
```
which means that after this change, `wh.allow` may be called with an empty domain string.

I'm not 100% sure what happens in that case tbh.  I doubt it's _damaging_, so it might be fine to do that, but it's worth figuring out.